### PR TITLE
Adding `alt` attribute to Icon components

### LINKED ISSUE #860: [BUG][ACCESSIBILITA'] Impossibile inserire la property "alt" nei componenti con icone
In tutti i componenti che renderizzano immagini e/o icon, come ad esempio "Icon" e "HeaderBrand" è impossibile inserire l'attributo "alt", rendendo qualsiasi applicazione che usa la libreria non compatibile con lo standard di accessibilità  WCAG2AA

### PR DESCRIPTION
**Pull request by dennybiasiolli**
Fixes #860 

#### PR Checklist

- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes (if applicable).
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable).
- [ ] I have added necessary documentation (if appropriate).

#### Short description of what this resolves:

This adds accessibility improvements for `Icon` and `HeaderBrand` components

#### Changes proposed in this Pull Request:

- stories(AutocompleteExample): removing a warning in tests

    `Parameter 'props' implicitly has an 'any' type`

- Icon: adding optional `alt` attribute in case of custom image

- HeaderBrand: adding `iconAlt` optional attribute

- Icon(assets): adding title element to svg icons

### CODE REVIEW

File: stories/Header/CompleteHeader/basic.tsx [hunk 1]
```diff
@@ -135,14 +135,17 @@ export class SlimHeader extends Component<
 }
 
 export class CenterHeader extends Component<
-  { theme?: 'light' | 'dark' | ''; iconName: string },
+  { theme?: 'light' | 'dark' | ''; iconName: string; iconAlt?: string },
```
Comment (dej611) [inline stories/Header/CompleteHeader/basic.tsx:138]: Can you pass an alt string also here to use this prop? https://github.com/italia/design-react-kit/pull/864/files#diff-5cd1ad6642e2f768061589327cb5b6e68233f59a39f80814afbc11b816a6c22aL250

**Review by dej611 — CHANGES_REQUESTED** [review]
> HeaderBrand: adding iconAlt optional attribute

I think this is ok as first pass but probably will add more follow ups for other components in this area.

The main structure of the PR is ok, but I'd recommend to remove all the `<title>` tags from the icons as suggested inline.

**Comment by dennybiasiolli** [comment]
Thanks for the comments, will work on them 😉

File: src/Icon/assets/ItArrowDown.tsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import * as React from 'react';
 export const component = (props: React.SVGProps<SVGSVGElement>) => {
   return (
     <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' {...props}>
+      <title>Arrow down icon</title>
```
[the remark this answers is not in the document]
Comment (dennybiasiolli) [reply src/Icon/assets/ItArrowDown.tsx:5]: Ok great, I'm trying to change them in that repo.
In the `1.x` branch there is already a `<title>` element in SVGs, but it's not "ported" to react components with `svgr` command, even if I add `--title-prop` parameter to the command.
Any suggestion here?
I can add `--title-prop` nonetheless, so we will have `title` props in every icon 😉

**Comment by dej611** [comment]
Thanks for the updates. I hope to check it tomorrow, and hopefully approve it 👍

**Comment by dennybiasiolli** [comment]
Thanks! Meanwhile I rebased this PR on top of current `master` branch 😉

**Review by dej611 — APPROVED** [review]
Dived into the SVGR title prop problem and fixed it into another branch.
This PR is approved and will send a follow up with the `title` fix later on. 👍

**Comment by dej611** [comment]
Thanks @dennybiasiolli for the PR! 🎉

**Pull request merged**